DOC-9205 - KV Operations dev-reorg (#2275)
couchbase/docs-server: modules/guides/pages/create-data.adoc

= Create Data
:description: How to create documents in Couchbase.
:page-pagination: full 
:page-topic-type: guide
:tabs:
:github: Click the GitHub button icon:github[] to view this code in context.

[abstract]
{description}

> Couchbase stores its xref:learn:data/data.adoc[data] as either JSON or Binary _documents_ with unique IDs(primary keys) by which they can be located.

Reviewer: I'd refer a space before the brackets?

```suggestion
Couchbase stores its xref:learn:data/data.adoc[data] as either JSON or Binary _documents_ with unique IDs (primary keys) by which they can be located.
```

Author: Good catch, there's a few more that I missed, will correct 👍 .

> Documents are organized and grouped in _buckets_ using xref:learn:data/scopes-and-collections.adoc[scopes and collections]. If a scope or collection is not provided when creating a document, Couchbase will automatically store data in the `_default` collection or `_default` scope.

Reviewer: Not a massive issue but we try to stick to [ventilated prose](https://docs.couchbase.com/home/contribute/pages.html#ventilate) — new line after each full stop:

```suggestion
Documents are organized and grouped in _buckets_ using xref:learn:data/scopes-and-collections.adoc[scopes and collections].
If a scope or collection is not provided when creating a document, Couchbase will automatically store data in the `_default` collection or `_default` scope.
```

TIP: We highly recommend that you provide user-defined scopes and collections to logically group your data.

To create a single document in Couchbase you can perform an `insert` operation.

[{tabs}]
====
cbc::
+ 
-- 
.Insert

> An example JSON document called `airport-123.json` is used here to represent our data.

Reviewer: Remember the how-to guides are instructions to enable users to carry out a task, not teaching / learning materials. I think it helps to rewrite the instructions as a numbered list.

Each procedure should start with "To do X", and each instruction in the procedure should be an imperative, "Do this!" — not "you can do this" or "we can do this".

```suggestion
.Inserting a single document
1. Store the document to insert in a correctly-structured JSON document.
For example, the JSON document called `airport-123.json` is used here to represent our data.
+
```

Need the `+` to ensure the code block stays attached to the list item

[source,json]
----
include::example$kv/airport-123.json[]
----

Using the `cbc create` command with `--mode insert` we create the document in the `travel-sample.inventory.airport` keyspace.

> [source,shell]

Reviewer: ```suggestion
2. Use the `cbc create` command with `--mode insert` to create the document.
Use the `--scope` and `--collection` parameters to specify the keyspace in which to store the document, and use `<` to specify the JSON document to insert.
+
[source,shell]
```

----
include::example$kv/kv-cbc.sh[tag=cbc-kv-insert]
----

.Result
[source,console]
----
airport-123          Stored. CAS=0x16b37e5081690000
                     SYNCTOKEN=829,51260943619833,92
----

> .Insert (with options)

Reviewer: I would be tempted to separate this out into a new procedure, "Inserting with options". I'll leave you to rework the following paragraphs as a procedure.

Author: Yeah, I was in two minds about this, will split into a new procedure. Seems nicer that way 👍
